cluster-autoscaler/core/scale_down.go

```diff
@@ -227,6 +227,36 @@ func calculateScaleDownGpusTotal(nodes []*apiv1.Node, cp cloudprovider.CloudProv
 	return result, nil
 }
 
+// skipCalculatingUnreadyNode checks if a node should be skipped calucating utilization.
+func (sd *ScaleDown) skipCalculatingUnreadyNode(node *apiv1.Node, timestamp time.Time) bool {
```

Comment:
I did not do a full review.
1. Just comment on this method. Name is very confusing and does not match description
2. The fact that predicate is modifying a map in ScaleDown is highly unexpected - I would expect this method to not have any side effects

Reply:
Sure. I will clean them up this weekend. I was stuck in some urgent work recently. Also waiting for feedbacks from @MaciekPytel

Reply:
@losipiuk Thanks. I rename it and make it more understandable. 
UnreadyNode map is to identify those nodes with unready resources explicitly. The original version doesn't have this map and we use a tricky way to skip unready nodes. This part can be removed. Both way work